subprojects/docs/src/docs/userguide/multi_project_builds.adoc

```diff
@@ -48,403 +48,660 @@ Configuration on demand mode attempts to configure only projects that are releva
 
 Let's look at some examples with the following project tree. This is a multi-project build with a root project named `water` and a subproject named `bluewhale`.
 
-++++
-<sample id="multiprojectFirstExample" dir="userguide/multiproject/firstExample/water" includeLocation="true" title="Multi-project tree - water &amp; bluewhale projects">
-                <layout>
-                    build.gradle
-                    settings.gradle
-                    bluewhale/
-                </layout>
-                <sourcefile file="settings.gradle"/>
-            </sample>
-++++
+=== Example: Multi-project tree - water &amp; bluewhale projects
+
+.Project layout
+----
+.
+├── bluewhale/
+├── build.gradle
+└── settings.gradle
+----
+
+[source.multi-language-sample,groovy]
+.settings.gradle
+----
+include::{samplesPath}/userguide/multiproject/firstExample/water/settings.gradle[]
+----
 
 And where is the build script for the `bluewhale` project? In Gradle build scripts are optional. Obviously for a single project build, a project without a build script doesn't make much sense. For multiproject builds the situation is different. Let's look at the build script for the `water` project and execute it:
 
-++++
-<sample id="multiprojectFirstExample" dir="userguide/multiproject/firstExample/water" title="Build script of water (parent) project">
-                <sourcefile file="build.gradle"/>
-                <output args="-q hello"/>
-            </sample>
-++++
+=== Example: Build script of water (parent) project
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/multiproject/firstExample/water/build.gradle[]
+----
+
+.Output of **`gradle -q hello`**
+----
+> gradle -q hello
+include::{samplesPath}/userguide/multiproject/firstExample/water/multiprojectFirstExample.out[]
+----
 
 Gradle allows you to access any project of the multi-project build from any build script. The Project API provides a method called `project()`, which takes a path as an argument and returns the Project object for this path. The capability to configure a project build from any build script we call _cross project configuration_. Gradle implements this via _configuration injection_.
 
 We are not that happy with the build script of the `water` project. It is inconvenient to add the task explicitly for every project. We can do better. Let's first add another project called `krill` to our multi-project build.
 
-++++
-<sample id="multiprojectAddKrill" dir="userguide/multiproject/addKrill/water" includeLocation="true" title="Multi-project tree - water, bluewhale  &amp; krill projects">
-                <layout>
-                    build.gradle
-                    settings.gradle
-                    bluewhale/
-                    krill/
-                </layout>
-                <sourcefile file="settings.gradle"/>
-            </sample>
-++++
+=== Example: Multi-project tree - water, bluewhale  &amp; krill projects
```

Comment:
:x: Missing note admonition about where to find the sample in the `-all` distribution, same as above